### ISSUE TITLE
Can't compile with TeXLive 2020

### ISSUE THREAD

**Issue by araujoms** [cgogolin/quantum-journal#111]
When I try to compile a paper with TeXLive 2020 - necessary because the arXiv has now updated to it - I get the following error:

```
! Undefined control sequence.
\enddocument ...firstofone \@afterenddocumenthook 
                                                  \deadcycles \z@ \@@end 
l.913 \end{document}
```

This seems to be related to the old ltxgrid bug. If I delete the following part of quantumarticle.cls I can compile the paper without any problems:

```
\ltx@ifpackageloaded{ltxgrid}{
	%repair what ltxutils has destroyed (see https://tex.stackexchange.com/questions/343856/biblatex-produces-incomplete-bcf-with-biber-backend) 
	\patchcmd\enddocument
	{\deadcycles}
	{\let\AfterEndDocument\@firstofone
		\@afterenddocumenthook
		\deadcycles}
	{}
	{\AtEndDocument{%
			\let\etb@@end\@@end
			\def\@@end{%
				\let\AfterEndDocument\@firstofone
				\@afterenddocumenthook
				\etb@@end}}}
}{}
```

I assume then that the new version of ltxgrid fixed whatever what it was doing wrong before.

**Comment by cgogolin** [comment]
Thanks for reporting!

I will look into this asap.
-- 
Christian Gogolin
http://www.cgogolin.de

**Comment by cgogolin** [comment]
I have trouble reproducing the problem. Could you please upload or send (info@quantum-journal.org) me the compete source files of the paper? Thanks!

**Comment by araujoms** [comment]
That's strange. The minimal working example is simply

```
\documentclass[a4paper]{quantumarticle}
\pdfoutput=1
\begin{document}
asdf
\end{document}
```
This is with a fully updated TeXLive 2020. If you want the complete source anyway it's [here](https://arxiv.org/abs/2005.13418).

**Comment by araujoms** [comment]
To be more precise, the problem seems to be triggered by the version 4.2d of ltxgrid (and revtex) that was just released. In the changelog of ltxgrid there is this line which seems relevant:
```
\changes{4.2d}{2020/09/19}{(PHO) Adapt \cs{document} and \cs{enddocument} hooks to the 2020-10-01 \LaTeX{} release.}%
```

**Comment by cgogolin** [comment]
First of all: Thanks for already investigating this further!

ltxgrid versioning is a mess. On the [ltxgrid page of CTAN](https://www.ctan.org/pkg/ltxgrid) it claims that the latest version is 4.1b. [Revtex is now at 4.2e](https://www.ctan.org/tex-archive/macros/latex/contrib/revtex) and the [ltxgrid file](https://ctan.kako-dev.de/macros/latex/contrib/revtex/ltxgrid.dtx) distributed as part of revtex claims to be version 4.2e in the @LaTeX-file block but only has a changelog up to 4.2d... Anyway...

Comparing old and new ltxgrid and ltxutil versions I have the impression that the [bug we are trying to solve with the above code snippet](ttps://tex.stackexchange.com/questions/343856/biblatex-produces-incomplete-bcf-with-biber-backend) was indeed fixed in the latest version of ltxgrid published on 2020/10/03.

With the following code (notice the `\@ifpackagelater`) I still get the benefit of the fix with the old version. Could you please try whether this solves the problem for you and makes things work correctly with the new version?
```
\ltx@ifpackageloaded{ltxgrid}{
  \@ifpackagelater{ltxgrid}{2020/10/03}
  {}% on 2020/10/03 ltxgrid 4.2e (or 4.2d?) was released, which not only fixes the bug we try to work around here but also the fix below causes an error from this version on. 
  {
	%repair what ltxutils has destroyed (see https://tex.stackexchange.com/questions/343856/biblatex-produces-incomplete-bcf-with-biber-backend) 
	\patchcmd\enddocument
	{\deadcycles}
	{\let\AfterEndDocument\@firstofone
		\@afterenddocumenthook
		\deadcycles}
	{}
	{\AtEndDocument{%
			\let\etb@@end\@@end
			\def\@@end{%
				\let\AfterEndDocument\@firstofone
				\@afterenddocumenthook
				\etb@@end}}}
  }
}{}
```
Thanks for your help!

**Comment by araujoms** [comment]
Yep, just tested here, it does work, thanks a lot!

It seems that revtex really doesn't care about people that reuse it's internals...

**Comment by cgogolin** [comment]
Fixed in f5cbb1dc05927d359b5e674458ce1cdb9dbe4017
